hasher-matcher-actioner/hmalib/common/actioner_models.py

```diff
@@ -43,49 +45,115 @@ class ActionRule:
+    after update_aws_config is called. To retrieve from AWS the correct
+    action performer for a label use the function ActionPerformerConfig.get_performer
+
+    NOTE all attributes must be aws serializable. see config._py_to_aws_field for serialization logic
+    """
+
+    action_label: ActionLabel
+
+    def perform_action(self, match_message: MatchMessage) -> None:
+        raise NotImplementedError
+
+
+@dataclass
+class WebhookActionPerformer(ActionPerformer):
+    url: TUrl
+
+    def perform_action(self, match_message: MatchMessage) -> None:
+        kwargs = {"data": match_message.to_sns_message()}
+        self.call(**kwargs)
+
+    def call(self, **kwargs) -> Response:
         raise NotImplementedError()
 
 
-class Post(HTTPRequest):
-    @staticmethod
-    def call(url: TUrl, **kwargs) -> Response:
+class WebhookPostActionPerformer(WebhookActionPerformer):
+    def call(self, **kwargs) -> Response:
         data = kwargs.get("data")
-        return post(url, data)
+        return post(self.url, data)
 
 
-class Get(HTTPRequest):
-    @staticmethod
-    def call(url: TUrl, **kwargs) -> Response:
-        return get(url)
+class WebhookGetActionPerformer(WebhookActionPerformer):
+    def call(self, **kwargs) -> Response:
+        return get(self.url)
 
 
-class Put(HTTPRequest):
-    @staticmethod
-    def call(url: TUrl, **kwargs) -> Response:
+class WebhookPutActionPerformer(WebhookActionPerformer):
+    def call(self, **kwargs) -> Response:
         data = kwargs.get("data")
-        return put(url, data)
+        return put(self.url, data)
 
 
-class Delete(HTTPRequest):
-    @staticmethod
-    def call(url: TUrl, **kwargs) -> Response:
-        return delete(url)
+class WebhookDeleteActionPerformer(WebhookActionPerformer):
+    def call(self, **kwargs) -> Response:
+        return delete(self.url)
 
 
-class ActionPerformer:
-    def perform_action(self, match_message: MatchMessage) -> None:
-        raise NotImplementedError()
+def get_all_subclasses_rec(recursive: t.Set[t.Type]) -> t.Set[t.Type]:
+    subclasses = {subclass for cls in recursive for subclass in cls.__subclasses__()}
+    union = recursive.union(subclasses)
+    if union != recursive:
+        return get_all_subclasses_rec(union)
+    return recursive
 
 
 @dataclass
-class WebhookActionPerformer(ActionPerformer):
-
-    request_type: t.Type[HTTPRequest]
-    url: TUrl
-
-    def perform_action(self, match_message: MatchMessage) -> None:
-        kwargs = {"url": self.url, "data": match_message.to_sns_message()}
-        self.request_type.call(**kwargs)
+class ActionPerformerConfig(config.HMAConfig):
+    """
+    An adapter to use the HMAConfig Class to store differnet types of
+    ActionPerformer configs. Since ActionPerformer configs are all unique by label but may
+    be of different types, we store the name of the concrete type and the
+    attributes of that type in AWS.
+
+    DO NOT CALL config.HMAConfig FUNCTIONS DIRECTLY
```

Comment:
What does this mean?

Reply:
Saying not to call the underlying class directly. I think should be
```
DO NOT CALL config FUNCTIONS DIRECTLY
```